docs/source/modin/performance.rst

```diff
@@ -0,0 +1,58 @@
+Performance Recommendations
+===========================
+
+This page contains recommendations to help improve performance when using the Snowpark pandas API.
+
+Caching Intermediate Results
+----------------------------
+Snowpark pandas uses a lazy paradigm - when operations are called on a Snowpark pandas object,
+a lazy operator graph is built up and executed only when an output operation is called (e.g. printing
+the data, or persisting it to a table in Snowflake). This paradigm mirrors the Snowpark DataFrame paradigm,
+and enables larger queries to be optimized using Snowflake's SQL Query Optimizer. Certain workloads; however,
+can generate large operator graphs that include repeated, computationally expensive, subgraphs, that can introduce
+recomputations that impact performance. Take the following code snippet as an example:
```

Comment:
```suggestion
can generate large operator graphs that include repeated computationally expensive subgraphs. Take the following code snippet as an example:
```

nit: " that can introduce recomputations that impact performance" makes the sentence a run-on, and I think it's redundant.